### PR TITLE
Iteration3/sleep

### PR DESCRIPTION
**Pull request by carlymclift**
#### What's this PR do?
Tests for methods written on the Sleep Class

#### Where should the reviewer start?
Sleep.js

#### How should this be manually tested?
npm

#### Any background context you want to provide?
The commented out methods are what I have been trying

#### What are the relevant tickets?
n/a

### CODE REVIEW

File: src/Sleep.js [hunk 1]
```diff
@@ -1,82 +1,132 @@
...
     }
-
+  
     findUserAverageSleep(dataset) {
-		let avSleep = dataset.reduce((acc, user) => {
-			acc += user.hoursSlept;
-			return acc;
-		}, 0)
-		return Math.round(avSleep / dataset.length);
+      let avSleep = dataset.reduce((acc, user) => {
```
Comment (st0nebreaker) [inline src/Sleep.js]: I think this can be a const

File: src/Sleep.js [hunk 2]
```diff
@@ -1,82 +1,132 @@
...
     }
-    
+  
     findUserAverageQuality(dataset) {
-		let avSleepQuality = dataset.reduce((acc, user) => {
-			acc += user.sleepQuality;
-			return acc;
-		}, 0)
-		return Math.round(avSleepQuality / dataset.length);
+      let avSleepQuality = dataset.reduce((acc, user) => {
```
Comment (st0nebreaker) [inline src/Sleep.js]: again think a const here

File: src/Sleep.js [hunk 3]
```diff
@@ -1,82 +1,132 @@
...
+      return Math.round(avSleepQuality / dataset.length);
     }
-
+  
     findUserSleepForDay(date) {
-		let sleepData = this.userSleepData.find(user => {
-			return user.date === date;
-		})
-		return sleepData.hoursSlept;
+      let sleepData = this.userSleepData.find(user => {
```
Comment (st0nebreaker) [inline src/Sleep.js]: +1 for const

File: test/Sleep-test.js [hunk 4]
```diff
@@ -31,11 +31,83 @@ describe('Sleep', () => {
...
+	it('should be an instance of Sleep', () => {
+		expect(userSleep).to.be.an.instanceof(Sleep);
 	});
 
 	it('should hold data from the user', () => {
 		expect(userSleep.id).to.equal(user1.id);
-	})
+    })
+    
+    it('should throw an error if instance of Sleep has no user arguments', () => {
```
Comment (st0nebreaker) [inline test/Sleep-test.js:42]: are these indented correctly underneath the it-statement? can't fully tell

**Review by st0nebreaker — CHANGES_REQUESTED** [review]
can you change some of those lets' to const' instead? i think i'm making this mistake too

File: test/Sleep-test.js (hunk 4 above)
Comment (carlymclift) [reply test/Sleep-test.js:42]: they should be! unless I've been doing it wrong. I think that are indented correctly

**Comment by st0nebreaker** [comment]
woo, thanks :) cool to see how we can request changes on open PR's too

**Pull request merged**